sh/src/helm-template

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env bash
+#
+# Render chart templates locally using helm template. Display the output objects as a Kubernetes List.
+
+set -eo pipefail
+
+NAME="name"
+CHART_PATH="chart_path"
+VALUES_PATH="values_path"
+USAGE=$(cat << END
+Render chart templates locally using helm template.
+Display the output objects as a Kubernetes List. If piped a Kubernetes List in
+addition to arguments then render the chart objects into the piped list,
+overwriting any chart objects that already exist in the list.
+
+Configured using a ConfigMap with the following keys:
+
+${NAME}: Name of helm chart.
+${CHART_PATH}: Chart templates directory.
+${VALUES_PATH}: [Optional] Path to values.yaml. Defaults to ${CHART_PATH}/values.yaml.
+
+Example:
+
+To expand a chart named 'my-chart' at '../path/to/helm/chart' using './values.yaml':
+
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-config
+  annotations:
+    config.k8s.io/function: |
+      container:
+        image:  gcr.io/kpt-functions/helm-template
+    config.kubernetes.io/local-config: "true"
+data:
+  ${NAME}: my-chart
+  ${CHART_PATH}: ../path/to/helm/chart
+  ${VALUES_PATH}: ./values.yaml
+END
+)
+
+err() {
+  echo "[$(date +'%Y-%m-%dT%H:%M:%S%z')]: $*" >&2
+}
+
+read_arguments() {
+  if [[ $# -le 1 ]] ; then
+    err "${USAGE}"
+    exit 1
+  fi
+
+  for arg in "$@"; do
+    case $arg in
+      "${NAME}="*) name="${arg#*=}" ;;
+      "${CHART_PATH}="*) chartPath="${arg#*=}" ;;
+      "${VALUES_PATH}="*) valuesPath="${arg#*=}" ;;
+      *)
+        err "${USAGE}"
+        exit 1
+        ;;
+    esac
+  done
+
+  if [[ -z $valuesPath ]]; then
+    valuesPath="${chartPath}/values.yaml"
+  fi
+}
+
+create_tmp() {
+  tmp=$(mktemp -d "/tmp/helm-template.XXXXXXXX")
+  # If there is input through stdin then write it to tmp
+  if [[ ! -t 0 ]]; then
+    cat "-" | kpt fn sink "$tmp"
+  fi
+}
+
+delete_tmp() {
+  rm -rf "$tmp"
+}
+
+expand_helm() {
+  # Overwrite files if they exist
+  if ! output="$(helm template "${name}" "${chartPath}" -f "${valuesPath}" --output-dir "${tmp}" )"; then
```

Comment:
Problem with wrapping helm cli like this is that you're preventing user from specify any other flag to template:

https://helm.sh/docs/helm/helm_template/

Instead, I think you want to only handle and redirect --output.

Reply:
I agree we prevent the user from specifying options. I'd rather check in this MVP first and address this in a separate change.

Reply:
That's fine. I would look into parsing $@ and passing the rest to Helm. This script should ideally only know about the output flag. In fact, in the most general form, this script can be reused for wrapping any CLIs like Helm template.

Reply:
I'm fine with following up on a separate PR